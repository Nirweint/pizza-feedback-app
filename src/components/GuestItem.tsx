import React from 'react';
import {ListItem, ListItemButton, ListItemIcon, ListItemText} from "@mui/material";
import {FeedbackForm} from "./FeedbackForm";
import {FormModal} from "./FormModal";
import {getLocalStorageState} from "../localStorage";
import {FeedbackType, GuestDietType} from "../types";
import {GuestFeedback} from "./GuestFeedback";

type GuestItemPropsType = {
    name: string
    eatsPizza: boolean
    guestDiet?: GuestDietType
    handleGuestsListClick: () => void
}

export const GuestItem = ({guestDiet, eatsPizza, name, handleGuestsListClick}: GuestItemPropsType) => {
    const feedbacksFromLocalStorage = getLocalStorageState<FeedbackType[]>('feedback', [])
    const currentGuestFeedback = feedbacksFromLocalStorage.filter(guest => guest.name === name)
    console.log(currentGuestFeedback)

    const [openModal, setOpenModal] = React.useState<boolean>(false);

    const handleGuestClick = () => {
        handleGuestsListClick()
        setOpenModal(true)
    }

    return (
        <ListItem disablePadding>
            <ListItemButton disabled={!eatsPizza} onClick={handleGuestClick}>
                {currentGuestFeedback.length !== 0 &&
				<ListItemIcon sx={{minWidth: 25}}>
					âœ…
				</ListItemIcon>
                }
                <ListItemText
                    primary={name}
                    style={{color: `${guestDiet && guestDiet.isVegan && eatsPizza ? 'green' : ''}`}}
                />
            </ListItemButton>
            {currentGuestFeedback.length === 0 ?
                <FormModal
                    setOpenModal={setOpenModal}
                    openModal={openModal}
                >
                    <FeedbackForm name={name} setOpenModal={setOpenModal}/>
                </FormModal>
                :
                <FormModal
                    setOpenModal={setOpenModal}
                    openModal={openModal}
                >
                   <GuestFeedback currentGuestFeedback={currentGuestFeedback[0]} setOpenModal={setOpenModal}/>
                </FormModal>
            }

        </ListItem>
    );
};