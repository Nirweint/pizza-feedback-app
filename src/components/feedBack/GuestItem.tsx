import React from "react";
import {
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText
} from "@mui/material";
import { FeedbackForm } from "./FeedbackForm";
import { FormModal } from './FormModal';
import {getLocalStorageState} from "../../localStorage";
import {FeedbackType, GuestDietType} from "../../types";
import {GuestFeedback} from "./GuestFeedback";

type GuestItemPropsType = {
  name: string;
  eatsPizza: boolean;
  guestDiet?: GuestDietType;
  handleGuestsListClick: () => void;
};

export const GuestItem = (props: GuestItemPropsType) => {
  const { guestDiet, eatsPizza, name, handleGuestsListClick } = props;

  const [openModal, setOpenModal] = React.useState<boolean>(false);

  const feedbacksFromLocalStorage = getLocalStorageState<FeedbackType[]>(
    "feedback",
    []
  );

  const currentGuestFeedback = feedbacksFromLocalStorage.filter(
    (guest) => guest.name === name
  );

  const handleGuestClick = () => {
    handleGuestsListClick();
    setOpenModal(true);
  };

  return (
    <ListItem disablePadding>
      <ListItemButton data-testid='name-wrapper' disabled={!eatsPizza} onClick={handleGuestClick}>
        {currentGuestFeedback.length !== 0 && (
          <ListItemIcon sx={{ minWidth: 25 }}>âœ…</ListItemIcon>
        )}
        <ListItemText
          data-testid='name-text'
          primary={name}
          style={{
            color: `${
              guestDiet && guestDiet.isVegan && eatsPizza ? "green" : ""
            }`
          }}
        />
      </ListItemButton>
      {currentGuestFeedback.length === 0 ? (
        <FormModal setOpenModal={setOpenModal} openModal={openModal}>
          <FeedbackForm name={name} setOpenModal={setOpenModal} />
        </FormModal>
      ) : (
        <FormModal setOpenModal={setOpenModal} openModal={openModal}>
          <GuestFeedback
            currentGuestFeedback={currentGuestFeedback[0]}
            setOpenModal={setOpenModal}
          />
        </FormModal>
      )}
    </ListItem>
  );
};
