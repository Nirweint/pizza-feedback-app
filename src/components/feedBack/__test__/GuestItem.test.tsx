import { render, screen } from '@testing-library/react';
import React from "react";
import {GuestItem} from "../GuestItem";

describe('GuestItem', () => {

  test('if person is vegan, his name must be green', () => {
    render(<GuestItem
      name='Alex'
      eatsPizza={true}
      guestDiet={{name: 'Alex', isVegan: true}}
      handleGuestsListClick={() => {
      }}/>);

    expect(screen.getByTestId('name-text')).toHaveStyle('color: green;')
  })


  test('if person dont eat pizza, his li must be disabled', () => {
    render(<GuestItem
      name='Alex'
      eatsPizza={false}
      guestDiet={{name: 'Alex', isVegan: true}}
      handleGuestsListClick={() => {
      }}/>);

    expect(screen.getByTestId('name-wrapper').getAttribute('aria-disabled')).toBe('true')
  })

  test('renders correctly', () => {
    const {asFragment} = render(<GuestItem
      name='Alex'
      eatsPizza={true}
      guestDiet={{name: 'Alex', isVegan: true}}
      handleGuestsListClick={() => {
      }}/>);

    expect(asFragment()).toMatchSnapshot();
  })

})
