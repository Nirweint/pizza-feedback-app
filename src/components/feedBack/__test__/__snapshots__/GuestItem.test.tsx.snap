// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestItem renders correctly 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
  >
    <div
      class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
      data-testid="name-wrapper"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        data-testid="name-text"
        style="color: green;"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Alex
        </span>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
    <div />
  </li>
</DocumentFragment>
`;
